src/algokit_utils/asset.py

```diff
@@ -0,0 +1,124 @@
+import logging
+from typing import TYPE_CHECKING
+
+from algosdk.atomic_transaction_composer import AtomicTransactionComposer, TransactionWithSigner
+from algosdk.transaction import AssetTransferTxn
+
+if TYPE_CHECKING:
+    from algosdk.v2client.algod import AlgodClient
+
+from algokit_utils.models import Account
+
+__all__ = ["opt_in", "opt_out"]
+logger = logging.getLogger(__name__)
+MAX_GROUP_SIZE = 16
+
+
+def ensure_asset_balance(algod_client: "AlgodClient", account: Account, asset_ids: list) -> None:
```

Comment:
are these exported to external use or only used internally by opt in and opt out? If not exported please make it private